Add code for converting Concept URIs to FHIR Codings and vice versa

gaurav opened the pull request:
This PR adds ConceptIRI.js, a class containing code for translating concept IRIs into and out of FHIR system/code pairs. It does this by requiring the [hl7-terminology](http://registry.fhir.org/package/hl7.terminology|3.0.0) NPM package, which contains JSON files describing HL7 NamingSystem and CodingSystems. This package retrieves stem IRI information from these JSON packages and uses them to perform this conversion. Note that IRI information isn't in the real hl7-terminology NPM package yet; instead, this code works with the fork on the UTG repository stored at https://github.com/HL7/UTG/pull/9. As we integrate stem IRIs into the NPM package, we will update this code to use the format we decide to use.

Discussed in https://github.com/w3c/hcls-fhir-rdf/issues/94

gaurav committed:
First stab at adding code to download FHIR examples.

gaurav committed:
Fixed Makefile.

gaurav committed:
Added basic classes and a test.

gaurav committed:
Converted ConceptIRI into a normal class.

gaurav committed:
Added code to extra concept IRIs and perform conversions.

gaurav committed:
Replaced ` in ` with `hasOwnProperty()`.

gaurav committed:
Added tests for additional prefixes and concept IRIs.

gaurav committed:
Updated tests for one failing data point.

gaurav committed:
Intermediate commit.

gaurav committed:
Added code for generating a summary.

gaurav committed:
Added generation of unique system/code pair lists for r4 and r5.

gaurav committed:
Added retus and resolved TSV generation.

gaurav committed:
Fixed bug, added display.

gaurav committed:
Fixed SNOMED URL.

gaurav committed:
Added MeSH example.

gaurav committed:
Added support for urn:ietf:rfc:3986 to indicate IRIs.

gaurav committed:
Added encodeURIComponent/decodeURIComponent, fixed bugs.

gaurav committed:
Uncommented full resolution.

gaurav committed:
Changed to baseURI as per change in HL7 Terminology.

gaurav committed:
Updated `baseURI` to `stemIRI`.

gaurav committed:
Replaced wget with curl.

gaurav committed:
Added smiley face as an example.

gaurav committed:
Used environmental variable to control test-all-examples.

gaurav committed:
First stab at testing surrogates and combined characters.

gaurav committed:
Fixed URL for new surrogates example.

ericprud commented:
This has a single conflict on [fhirlib/package.json](https://github.com/ericprud/fhircat-playground/pull/1/files#diff-55a39beedabb41894b821d27c8a4255e6a7870a700c06d7a62e123514e6763c6R26)
``` diff
@@@ -21,7 -20,7 +21,11 @@@
      "@shexjs/node": "^1.0.0-alpha.15",
      "@shexjs/util": "^1.0.0-alpha.15",
      "@shexjs/validator": "^1.0.0-alpha.17",
-     "commander": "^8.3.0",
+     "hl7.terminology": "^3.0.0",
      "jsonld": "^5.2.0",
      "n3": "^1.11.2"
    }
```
If you're probably the only person who's checked out your branch, you can rebase on `fhircat/json-ld.org`.

Commit:
~ update to latest Observation content model

Commit:
+ rdvch for JSON-LD contexts

Commit:
- fhirlib/test/fhir/medreq-{expected,min-trimmed}.shexj

Commit:
+ rdvch for ShExJ

Commit:
+ .gitignore += EGP/ /.idea/

Commit:
+ support for R in RDVch

Commit:
+ marginal RDVch support for r in playground

Commit:
+ RDVch support for d in playground

Commit:
+ #rdvchStatus

Commit:
+ FhirPreprocessors += RDVch support for h

dbooth-boston committed:
Trigger rebuild

Commit:
~ simplify editRdvchStatusText() invocations

Commit:
Merge branch 'main' of github.com:fhircat/fhir-rdf-playground into main

Commit:
~ fix program flow from markup to activeTab

Commit:
- top-level json-ld.org index file

Commit:
~ move /fhir-rdf-playground/playground/ to /fhir-rdf-playground/

Commit:
~ move #processing-errors after #final-jsonld

Commit:
+ ShExJGenerator += axes.h

Commit:
+ fallbackToN3Writer()

Commit:
+ extractPrefixesAndBaseFromJsonld

dbooth-boston committed:
Updated usage instructions

Commit:
~ fix z-worker

Commit:
Merge branch 'main' of github.com:fhircat/fhir-rdf-playground into main

Commit:
+ rdf:Collections and Hoist scalars

FhirJsonLdContextGenerator += @container @list
FhirShExJGenerator += OneOrMore_ lists
FhirPreprocessors += OneOrMore_ lists

Commit:
~ playground: fix permalinks

Commit:
~ first pass and optimal nestable list

Commit:
+ nested lists

Commit:
+ Nesting classes

Commit:
~ ...C. renders collections

Commit:
~ fix indentation of predicateObjectList

Commit:
Merge pull request #5 from fhircat/nested-list

+ render RDF Collections as nested lists

Commit:
~ move `globalThis.Contexts = {}`

Commit:
~ improve NestedWriter whitespace

Commit:
~ simplify propertyOverrides and fix Narrative.div

Commit:
~ shorten literals in test examples

Commit:
~ shorten literals in playground examples

these should really be dynamically loaded. later...

Commit:
Merge pull request #6 from fhircat/shortenExampleLiterals

Shorten example literals

Commit:
+ OneOrMore_code_AND_filter-operator

keeps CodeSystem.rdVCH from barfing in playground but it loses scalars
directly in Collections.

Commit:
~ NestedWriter: fix missing list element

Commit:
~ fix `fhir:value "generated", "generated"`

Commit:
+ dynamic loading of example manifest

Can now add a `manifestURL=` search parameter. Default is:
https://fhircat.github.io/fhir-rdf-playground/?manifestURL=playground/examples-manifest.json

Commit:
+ (giant) content model for contained resources

closes #11

Commit:
~ catch up

gaurav committed:
Installing NPM package hl7.terminology from simplifier.net.

gaurav committed:
Added lodash to dependencies.

gaurav committed:
Moved lodash to devDependencies.

gaurav committed:
Replaced stemIRI with IRIstem.

gaurav committed:
Updated NameSystem for country code.

gaurav committed:
Better documented ConceptIRI.test.js.

gaurav committed:
Attempted to implement the algorithm in the proposal.

Q&D solution for UTF-8 escaping.

gaurav committed:
Updated concept IRIs under the new rules from the proposal.

gaurav committed:
Documented the oddity.

gaurav committed:
First stab at fixing the UTF8 escape code to support UTF16 properly.

gaurav committed:
Fix code... maybe.

gaurav committed:
Readded tests for failing characters.

gaurav committed:
Reorganized code to complain about invalid characters.

gaurav committed:
Updated test to test error message.

gaurav committed:
Added test for codes with spaces.

gaurav commented:
Superceded by PR https://github.com/fhircat/fhir-rdf-playground/pull/12, closing.
